### PR TITLE
test: migrated request-id.test.js from tap to node:test

### PR DESCRIPTION
**Pull request by Tony133**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

Proposal:

  - migrated `request-id.test.js` from `tap` to `node:test` 🔥

### CODE REVIEW

File: test/request-id.test.js [hunk 1]
```diff
@@ -64,13 +63,15 @@ t.test('The request id header key can be customized', (t) => {
         'my-custom-request-id': REQUEST_ID
       }
     }, (err, response, body) => {
-      t.error(err)
-      t.equal(body.toString(), `{"id":"${REQUEST_ID}"}`)
+      t.assert.ifError(err)
+      t.assert.strictEqual(body.toString(), `{"id":"${REQUEST_ID}"}`)
+      fastify.close()
```
Comment (dancastillo) [inline test/request-id.test.js:68]: does having `t.after(() => fastify.close())` work here?

File: test/request-id.test.js [hunk 2]
```diff
@@ -94,13 +94,15 @@ t.test('The request id header key can be customized', (t) => {
         'MY-CUSTOM-REQUEST-ID': REQUEST_ID
       }
     }, (err, response, body) => {
-      t.error(err)
-      t.equal(body.toString(), `{"id":"${REQUEST_ID}"}`)
+      t.assert.ifError(err)
+      t.assert.strictEqual(body.toString(), `{"id":"${REQUEST_ID}"}`)
+      fastify.close()
```
Comment (dancastillo) [inline test/request-id.test.js:99]: same as above: does having `t.after(() => fastify.close())` work here?

Comment (Tony133) [reply test/request-id.test.js]: yes, see commit: [a124c86](https://github.com/fastify/fastify/pull/5858/commits/a124c86fb57622f415ef59f7d2ee3b7e2c1d4f63)

File: test/request-id.test.js (hunk 1 above)
Comment (Tony133) [reply test/request-id.test.js]: yes, see commit: [a124c86](https://github.com/fastify/fastify/pull/5858/commits/a124c86fb57622f415ef59f7d2ee3b7e2c1d4f63)

Comment (Tony133) [reply test/request-id.test.js]: yes, see commit: [a124c86](https://github.com/fastify/fastify/pull/5858/commits/a124c86fb57622f415ef59f7d2ee3b7e2c1d4f63)

**Review by gurgunday — APPROVED** [review]

**Review by dancastillo — APPROVED** [review]
LGTM

**Review by Eomm — APPROVED** [review]

**Pull request merged**